packages/contract/lib/contract.js

```diff
@@ -20,13 +20,6 @@ if (typeof Web3 === "object" && Object.keys(Web3).length === 0) {
     var instance = this;
     var constructor = instance.constructor;
 
```

Comment:
Will this be a breaking change removing this functionality? I'm not sure if it is useful for external use or if it is even documented.

Reply:
Good question. It's at least an internal/module-specific breaking change. I think the benefit of normalizing the documented & exposed API behavior outweighs the rare possible usage of something like `const newInstance = new SimpleStorage(SimpleStorage.address)` vs `const newInstance = SimpleStorage.at(SimpleStorage.address)`.

I don't think we really want or expect people to do the former.

Reply:
It is a breaking change! Also, there's https://en.m.wikipedia.org/wiki/Robustness_principle, which says to be liberal in what inputs you accept.

I am against removing this.

Reply:
Good point. Will rework the code to be non-breaking.

Reply:
By the way, I'm not against your renaming the input to this function to more accurately describe what it is. When the argument says "contract" I expect it to be a contract, not a string representation of the address. This is the kind of stuff that makes the code super confusing and hard to read. :)